orttraining/orttraining/python/ort_trainer.py

```diff
@@ -681,6 +697,18 @@ def _init_session(self):
             return
 
         self._verify_fully_optimized_model(self.onnx_model_)
+
+        if self.save_model_dir or self.run_symbolic_shape_infer:
```

Comment:
What scenarios are being unlocked by this change on ORTTrainer? Should this be done for both training and evaluation? InferenceSession is handled by https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/session.py#L158

Reply:
BART needs this change, so most tensors could be statically planned. Inference does not rely on shape info for memory pattern though. ([https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/session_state.cc#L313-L352](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/session_state.cc#L313-L352))

---
In reply to: [488801960](https://github.com/microsoft/onnxruntime/pull/5107#discussion_r488801960) [](ancestors = 488801960)